## Primera exploración


In [26]:
# Tratamiento de datos
import numpy as np
import pandas as pd

# Gráficos
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.graph_objects as go

# Estadísticas
import scipy
 
# mis cosas
import sys
import os

# Añado el directorio padre (del que está este notebook) a sys.path
sys.path.append(os.path.abspath('../'))
from scripts.utils_agv import ini_inspec

#from Prof.utils import create_summary_table, descriptive_statistics, plot_variable_distribution, grafico_combinaciones

(primeramente renombramos el Dataframe de trabajo, a la vez que hacemos una copia para no tener que cargarlo de nuevo si lo "estropeamos")

In [2]:
viviendas_Madrid = pd.read_csv('../data/raw/idealista Madrid.csv')
df = viviendas_Madrid.copy()

In [103]:
df.head(12)

url  \
0   https://www.idealista.com/inmueble/104027174/   
1   https://www.idealista.com/inmueble/102321942/   
2   https://www.idealista.com/inmueble/103334142/   
3   https://www.idealista.com/inmueble/104161987/   
4   https://www.idealista.com/inmueble/103989666/   
5   https://www.idealista.com/inmueble/103963452/   
6   https://www.idealista.com/inmueble/103363210/   
7   https://www.idealista.com/inmueble/103938653/   
8   https://www.idealista.com/inmueble/103789810/   
9   https://www.idealista.com/inmueble/103653079/   
10  https://www.idealista.com/inmueble/103876000/   
11  https://www.idealista.com/inmueble/101184519/   

                                           listingUrl  \
0   https://www.idealista.com/venta-viviendas/madr...   
1   https://www.idealista.com/venta-viviendas/madr...   
2   https://www.idealista.com/venta-viviendas/madr...   
3   https://www.idealista.com/venta-viviendas/madr...   
4   https://www.idealista.com/venta-viviendas/madr...   
5   https://www.idealista.com/venta-viviendas/madr...   
6   https://www.idealista.com/venta-viviendas/madr...   
7   https://www.idealista.com/venta-viviendas/madr...   
8   https://www.idealista.com/venta-viviendas/madr...   
9   https://www.idealista.com/venta-viviendas/madr...   
10  https://www.idealista.com/venta-viviendas/madr...   
11  https://www.idealista.com/venta-viviendas/madr...   

                                             title         id    price  baths  \
0             Piso en venta en calle de Villanueva  104027174  1920000      3   
1        Piso en venta en calle de Núñez de Balboa  102321942  1995000      3   
2             Piso en venta en Conde Orgaz-Piovera  103334142  1300000      3   
3                    Piso en venta en Nueva España  104161987  1650000      3   
4         Piso en venta en calle de Claudio Coello  103989666  1590750      3   
5       Piso en venta en avenida del General Perón  103963452  1590000      4   
6   Piso en venta en calle del Príncipe de Vergara  103363210  5700000      5   
7                           Piso en venta en Ayala  103938653  2190000      4   
8                  Ático en venta en calle de Goya  103789810  1190000      2   
9               Piso en venta en avenida del Talgo  103653079   675000      2   
10      Piso en venta en calle del Conde de Aranda  103876000  3950000      5   
11                    Piso en venta en calle Mayor  101184519   752248      2   

    rooms  sqft                                        description  \
0       3   183  Residencia única con acabados de la más alta c...   
1       3   170  Preciosa reforma a estrenar, con terrazas y ga...   
2       4   270  DE SALAS CONSULTORES INMOBILIARIOS MAV02679 le...   
3       3   248  Gilmar Real Estate vende espectacular piso en ...   
4       3   116  Descubre esta impresionante vivienda exterior ...   
5       5   360  Walter Haus presenta este excepcional piso ext...   
6       4   305  DIZA consultores les presenta vivienda de lujo...   
7       4   183  Nos complace presentar esta espectacular vivie...   
8       2   112  DIZA Consultores, les presenta ático reformado...   
9       3   153  ¡Bajo duplex de 150m2, con jardín de 50m2, 3 d...   
10      4   291  DIZA consultores les presenta vivienda reforma...   
11      3   132  Luminoso piso exterior en calle MayorEngel & V...   

                   address typology  \
0        Recoletos, Madrid    Pisos   
1       Castellana, Madrid    Pisos   
2        Hortaleza, Madrid    Pisos   
3        Chamartín, Madrid    Pisos   
4        Recoletos, Madrid    Pisos   
5   Cuatro Caminos, Madrid    Pisos   
6        Recoletos, Madrid    Pisos   
7             Goya, Madrid    Pisos   
8             Goya, Madrid    Pisos   
9          Aravaca, Madrid    Pisos   
10       Recoletos, Madrid    Pisos   
11             Sol, Madrid    Pisos   

                           advertiserProfessionalName  \
0                                     Promora Madrid    
1                          

In [ ]:
ini_inspec(df)

In [105]:
print(df['typology'].unique())
print(df['address'].unique())

#de momento, no parece interesante la cardinalidad del anunciante para hacer hipotesis
#print(df['advertiserProfessionalName'].unique())
#print(df['advertiserName'].unique())

['Pisos' 'Independientes']
['Recoletos, Madrid' 'Castellana, Madrid' 'Hortaleza, Madrid'
 'Chamartín, Madrid' 'Cuatro Caminos, Madrid' 'Goya, Madrid'
 'Aravaca, Madrid' 'Sol, Madrid' 'Bernabéu-Hispanoamérica, Madrid'
 'Vallehermoso, Madrid' 'Huertas-Cortes, Madrid' 'Chamberí, Madrid'
 'Nueva España, Madrid' 'Conde Orgaz-Piovera, Madrid'
 'Chueca-Justicia, Madrid' 'Argüelles, Madrid'
 'Nuevos Ministerios-Ríos Rosas, Madrid' 'Centro, Madrid'
 'Moncloa, Madrid' 'Retiro, Madrid' 'Barrio de Salamanca, Madrid'
 'Peñagrande, Madrid' 'Mirasierra, Madrid' 'Fuencarral, Madrid'
 'Comillas, Madrid' 'Malasaña-Universidad, Madrid'
 'Palos de Moguer, Madrid' 'San Blas, Madrid' 'Prosperidad, Madrid'
 'Bellas Vistas, Madrid' 'Niño Jesús, Madrid' 'El Viso, Madrid'
 'Palacio, Madrid' 'Lista, Madrid' 'Ciudad Lineal, Madrid'
 'Lavapiés-Embajadores, Madrid' 'Salvador, Madrid' 'Almagro, Madrid'
 'Fuente del Berro, Madrid' 'Gaztambide, Madrid' 'Acacias, Madrid'
 'Arapiles, Madrid' 'Guindalera, Madrid' 'Ciudad

Veamos el conteo de cuantos unicos hay:

In [104]:
print(len(df['address'].unique())) 
print(df['address'].nunique())

105
105


Veamos si todos los únicos contienen "Madrid", o alguno es de otra ciudad:

In [100]:
pd.Series(df['address'].unique()).str.contains('Madrid')
#len(pd.Series(df['address'].unique()).str.contains('Madrid'))

0      True
1      True
2      True
3      True
4      True
       ... 
100    True
101    True
102    True
103    True
104    True
Length: 105, dtype: bool

Confirmamos que todas las direcciones corresponden a Madrid. 

Voy a hacer una investigación de distritos y barrios con un par de tablas que he descargado de
https://datos.gob.es/en/catalogo?publisher_display_name=Ayuntamiento+de+Madrid





In [ ]:
distritos_Madrid = pd.read_csv('../data/external/Madrid_Distritos.csv', sep=';')
distritos_Madrid


In [ ]:
dM = distritos_Madrid.copy().set_index('OBJECTID')
dM

Hasta aquí el DF de los 21 distritos de Madrid, limpiado del indice

In [ ]:
distritos_datos = pd.read_csv('../data/external/distritos_municipio_madrid.csv',sep=';')
distritos_datos

In [ ]:
dMd = distritos_datos.copy()
dMd

Y aquí un DF con los datos de densidad, extensión, etc de los 21 distritos de Madrid. Realizada copia renombrado cómodo.


Voy a hacer los mismo con los barrios, que son 131

In [ ]:
barrios_Madrid = pd.read_csv('../data/external/Madrid_Barrios.csv', sep=';')
barrios_Madrid

In [101]:
bM = barrios_Madrid.copy().set_index('CODDIS')
bM

NOMDIS  COD_BAR                      NOMBRE  COD_DIS_TX  \
CODDIS                                                             
1        Centro       12                 Embajadores           1   
1        Centro       13                      Cortes           1   
1        Centro       16                         Sol           1   
1        Centro       11                     Palacio           1   
1        Centro       14                    Justicia           1   
...         ...      ...                         ...         ...   
21      Barajas      211            Alameda de Osuna          21   
21      Barajas      215                  Corralejos          21   
21      Barajas      213  Casco Histórico de Barajas          21   
21      Barajas      214                       Timón          21   
21      Barajas      212                  Aeropuerto          21   

                        BARRIO_MAY  COD_DISBAR  NUM_BAR  \
CODDIS                                                    
1                      EMBAJADORES          12        2   
1                           CORTES          13        3   
1                              SOL          16        6   
1                          PALACIO          11        1   
1                         JUSTICIA          14        4   
...                            ...         ...      ...   
21                ALAMEDA DE OSUNA         211        1   
21                      CORRALEJOS         215        5   
21      CASCO HISTORICO DE BARAJAS         213        3   
21                           TIMON         214        4   
21                      AEROPUERTO         212        2   

                         BARRIO_MT COD_DISB  
CODDIS                                       
1                      EMBAJADORES   01-feb  
1                           CORTES   01-mar  
1                              SOL   01-jun  
1                          PALACIO   01-ene  
1                         JUSTICIA   01-abr  
...                            ...      ...  
21                ALAMEDA DE OSUNA   21-ene  
21                      CORRALEJOS   21-may  
21      CASCO HISTÓRICO DE BARAJAS   21-mar  
21                           TIMÓN   21-abr  
21                      AEROPUERTO   21-feb  

[131 rows x 9 columns]

In [ ]:
barrios_datos = pd.read_csv('../data/external/barrios_municipio_madrid.csv',sep=';',encoding='latin1')
barrios_datos

In [102]:
bMd = barrios_datos.copy()
bMd

barrio_codigo                       barrio_nombre  distrito_codigo  \
0           796011                             Palacio            79601   
1           796012                         Embajadores            79601   
2           796013                              Cortes            79601   
3           796014                            Justicia            79601   
4           796015                         Universidad            79601   
..             ...                                 ...              ...   
123         796211                    Alameda de Osuna            79621   
124         796212                          Aeropuerto            79621   
125         796213          Casco Histórico de Barajas            79621   
126         796214                               Timón            79621   
127         796215                          Corralejos            79621   

    distrito_nombre  municipio_codigo municipio_nombre  superficie_km2  \
0           Centro                796           Madrid            1.46   
1           Centro                796           Madrid            1.03   
2           Centro                796           Madrid            0.59   
3           Centro                796           Madrid            0.75   
4           Centro                796           Madrid            0.93   
..              ...               ...              ...             ...   
123        Barajas                796           Madrid            1.98   
124        Barajas                796           Madrid           19.87   
125        Barajas                796           Madrid            0.64   
126        Barajas                796           Madrid           16.40   
127        Barajas                796           Madrid            4.67   

     densidad_por_km2  
0            15323.29  
1            43345.63  
2            17850.85  
3            21866.67  
4            33051.61  
..                ...  
123           9795.45  
124             89.88  
125          11415.62  
126            672.01  
127           1577.09  

[128 rows x 8 columns]

Hasta aquí el DF de los 131 barrios de Madrid (realizada copia renombrado cómodo), limpiado del indice, usado como índice el Codigo de Distrito

------------

Voy a comprobar si el contenido de 'adress' corresponde siempre con un barrio

### Conclusiones

Observamos que tenemos 915 registros (pisos), con 13 columnas, entre las que observamos:

- Url y Listing URL parecen "duplicadas". En todo caso, para nuestro estudio no tienen más relevancia que poder verificar, si fuera el caso, el origen y veracidad de los datos que tenemos. 
- Las dos últimas, advertiserProfessionalName y advertiserName, de nuevo, parecen duplicadas y carecen de interes a priori
- La columna id constituye un indice y puede funcionar como tal, eliminando la columna 0 de indice del dataframe
- nombres de columnas en inglés, que consideraré si me es más operativo traducir
- me llama la atención que title (al igual que Adress, Tipology, etc) no sea una cadena de texto sino un objeto
- observo que en 'description' hay un valor único que habrá que estudiar
- el número de 'Address' unicos es 105, lo cual no son los distritos de Madrid, ni los barrios. Pero sí hemos comprobado que los 105 valores únicos de Address son todos de la ciudad de Madrid




Madrid tiene 21 distritos y 131 barrios. Aunque es este DF no tenemos ninguno de ambas columnas, sería interesante poder disponer al menos de la columna distrito para cada inmueble, para poder hacer hipótesis y análisis.

![](Mapa-Madrid-Distritos-Barrios.jpg)